### PR TITLE
#14686

### PR DESCRIPTION
**Pull request by chemsnouioua**
#14686

### CODE REVIEW

File: backend/de.metas.business/src/main/java-legacy/org/compiere/model/CalloutOrder.java [hunk 1]
```diff
@@ -331,13 +331,18 @@ public String bPartnerLocation(final ICalloutField calloutField)
 			orderBL.setBPLocation(order, bpartner);
 		}
 
-		if (!orderBL.setBillLocation(order))
+
+		if(order.getBill_Location_ID() <= 0)
 		{
```
Comment (cristinamghita) [inline backend/de.metas.business/src/main/java-legacy/org/compiere/model/CalloutOrder.java:336]: @chemsnouioua I am not ok with this. Pls debug here `de.metas.order.impl.OrderBL#setBillLocation` why this set to null and to the proper address.

**Review by cristinamghita — CHANGES_REQUESTED** [review]
See my comment

File: backend/de.metas.business/src/main/java-legacy/org/compiere/model/CalloutOrder.java (hunk 1 above)
Comment (chemsnouioua) [reply backend/de.metas.business/src/main/java-legacy/org/compiere/model/CalloutOrder.java:336]: @cristinamghita 
After the change of the delivery address, the callout  `org.compiere.model.CalloutOrder#bPartnerLocation` starts.
When de.metas.order.impl.OrderBL#setBillLocation whe do not calculate nor pass the `BillToAddress` :eye: screenshot
image

imho, my fix is still valid, try to get it only when its  null

Comment (cristinamghita) [reply backend/de.metas.business/src/main/java-legacy/org/compiere/model/CalloutOrder.java:336]: @chemsnouioua 
Yes, we do it. Here: `de.metas.document.location.adapter.IDocumentBillLocationAdapter#setFrom(de.metas.document.location.DocumentLocation)`

Comment (chemsnouioua) [reply backend/de.metas.business/src/main/java-legacy/org/compiere/model/CalloutOrder.java:336]: replied with DM

Comment (cristinamghita) [reply backend/de.metas.business/src/main/java-legacy/org/compiere/model/CalloutOrder.java:336]: @chemsnouioua See my fix. Please, test it. I did not!

**Review by cristinamghita — CHANGES_REQUESTED** [review]
Dev test it!

**Review by chemsnouioua — COMMENTED** [review]
works fine, thx @cristinamghita

File: backend/de.metas.business/src/main/java-legacy/org/compiere/model/CalloutOrder.java (hunk 1 above)
Comment (chemsnouioua) [reply backend/de.metas.business/src/main/java-legacy/org/compiere/model/CalloutOrder.java:336]: works fine, thx @cristinamghita

**Pull request merged**